src/firmware/app/control/trajectory_planner.c

```diff
@@ -0,0 +1,356 @@
+}
+
+
+void app_trajectory_planner_interpolateConstantTimeTrajectorySegmentation(
+    Trajectory_t constant_period_trajectory[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    Trajectory_t variable_time_trajectory[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    const float interpolation_period)
+{
+    // The first point is the same for each trajectory
+    constant_period_trajectory->trajectory_elements[0].time =
+        variable_time_trajectory->trajectory_elements[0].time;
+    constant_period_trajectory->trajectory_elements[0].position =
+        variable_time_trajectory->trajectory_elements[0].position;
+
+    // Keep track of the current time we are searching for in the constant arclength
+    // trajectory
+    unsigned int time_periods = 1;
+    float trajectory_time     = interpolation_period * time_periods;
+
+    // Loop until we find a value JUST larger than the expected
+    // Check the element prior and perform linear interpolation
+    for (unsigned int i = 1; i < variable_time_trajectory->num_elements; i++)
+    {
+        while (variable_time_trajectory->trajectory_elements[i].time >= trajectory_time)
+        {
+            // Perform a linear interpolation
+            const float delta_time =
+                variable_time_trajectory->trajectory_elements[i].time -
+                variable_time_trajectory->trajectory_elements[i - 1].time;
+            const float delta_x =
+                variable_time_trajectory->trajectory_elements[i].position.x -
+                variable_time_trajectory->trajectory_elements[i - 1].position.x;
+            const float delta_y =
+                variable_time_trajectory->trajectory_elements[i].position.y -
+                variable_time_trajectory->trajectory_elements[i - 1].position.y;
+
+            const float slope_x = delta_x / delta_time;
+            const float slope_y = delta_y / delta_time;
+
+            const float interpolated_x =
+                slope_x * (trajectory_time -
+                           variable_time_trajectory->trajectory_elements[i - 1].time) +
+                variable_time_trajectory->trajectory_elements[i - 1].position.x;
+            const float interpolated_y =
+                slope_y * (trajectory_time -
+                           variable_time_trajectory->trajectory_elements[i - 1].time) +
+                variable_time_trajectory->trajectory_elements[i - 1].position.y;
+
+            constant_period_trajectory->trajectory_elements[time_periods].position.x =
+                interpolated_x;
+            constant_period_trajectory->trajectory_elements[time_periods].position.y =
+                interpolated_y;
+            constant_period_trajectory->trajectory_elements[time_periods].time =
+                trajectory_time;
+
+            time_periods++;
+            trajectory_time = interpolation_period * time_periods;
+        }
+    }
+
+    constant_period_trajectory->num_elements = time_periods;
+}
+
+
+float app_trajectory_planner_getTotalArcLength(
```

Comment:
This seems like a function that should be implemented and tested in `polynomial_2d`. @garethellis0 thoughts?

Reply:
Moved to math/polynomial2d and added unit test